### PR TITLE
fix(excel): parsing of date values and reduced memory footprint. closes #3355

### LINKED ISSUE #3355: Feature Request - Improved Handling of Date Fields in XLSX Uploads
**Description:**
When uploading XLSX files containing date fields, errors are encountered if the corresponding Excel cell is set as a date. This issue does not occur when using text uploads.

**Issue Details:**

Problem: Errors arise during the XLSX upload process when the Excel cell is configured as a date field.
Expected Behavior: Smooth handling of date fields during XLSX uploads, similar to text uploads.
Impact: Users face difficulties in importing XLSX files with date fields, leading to inconvenience and potential data integrity issues.

**Screenshots**
Scherm­afbeelding 2024-02-05 om 11 35 37
***XLSX cell settings***
Scherm­afbeelding 2024-02-05 om 13 54 56
***Error given***
Scherm­afbeelding 2024-02-05 om 11 37 24
***workaround, set cell to text, (before entering date)***

**Additional context**
Added xlsx examples, second example uploads. Other one does not
[pet store_Order_DateTest.xlsx](https://github.com/molgenis/molgenis-emx2/files/14166006/pet.store_Order_DateTest.xlsx)
[pet store_Order_DateTest2.xlsx](https://github.com/molgenis/molgenis-emx2/files/14166008/pet.store_Order_DateTest2.xlsx)

### PR DESCRIPTION
**Pull request by mswertz**
closes #3355 

In addition
* in light of server crashes, converted from loading all rows in memory to iterating to hopefully reduce memory footprint
* upgraded to maintained streaming excel reader library (is continuation from lib we used before)

how to test:
- see #3355
- decrease in memory usage hard to measure

### CODE REVIEW

**Review by chinook25 — COMMENTED** [review]
Nice, it works with the excel available formats for me. Just got some code comments/questions.

File: backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java [hunk 1]
```diff
@@ -117,132 +123,164 @@ private void writeRowsToSheet(
...
+
+  @Override
+  public boolean containsTable(String name) {
+    return this.tableNames().contains(name);
+  }
+
+  public class WorkbookRowIterable implements Iterable<Row>, AutoCloseable {
+    private final Workbook workbook;
+    private final InputStream is;
+    private final String tableName;
```
[the remark this answers is not in the document]
Comment (mswertz) [reply backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java:144]: when the value is set in the constructor than that variable can be final
(it is actually a sonar warning if you don't make it final while you never change it, they find that a smell apparantly)

File: backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java [hunk 2]
```diff
@@ -117,132 +123,164 @@ private void writeRowsToSheet(
...
-  @Override
-  public List<Row> readTable(String name) {
-    if (this.cache == null) {
-      this.cache();
-    }
-    if (!this.cache.containsKey(name)) {
-      throw new MolgenisException(
-          "Import failed: Table with name " + name + " not found in Excel file");
+    @Override
+    public void close() {
```
[the remark this answers is not in the document]
Comment (mswertz) [reply backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java:171]: this is the standard interface of java we use

File: backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java [hunk 3]
```diff
@@ -117,132 +123,164 @@ private void writeRowsToSheet(
...
+        if (iterator.hasNext()) {
+          for (Cell cell : iterator.next()) {
+            String value = cell.getStringCellValue();
+            if (value != null) {
+              value = value.trim();
+            }
+            columnNames.put(cell.getColumnIndex(), value);
+          }
+        } else {
+          container.close();
```
[the remark this answers is not in the document]
Comment (mswertz) [reply backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/tablestore/TableStoreForXlsxFile.java:208]: java says we should do this to not have dangling file connection open. There is a potential that connection is never closed wen processing is in error state. We might need to change the interface to not be an 'iterator' but a 'forEach' so we can catch errors and then close also. But lets see if that is necessary.

**Review by chinook25 — APPROVED** [review]
Code looks fine to me now :D

**Review by Fernananas — APPROVED** [review]
I tested with excel (attached), all formats work!
[pet store_Pet1712058020695.xlsx](https://github.com/molgenis/molgenis-emx2/files/14835985/pet.store_Pet1712058020695.xlsx)

**Pull request merged**